runtime/src/accounts.rs

```diff
@@ -887,7 +871,26 @@ impl Accounts {
                             (false, true, Some((_, _, Some(fee_payer_account)))) => {
                                 *account = fee_payer_account.clone();
                             }
-                            _ => unreachable!(),
+                            (is_nonce_account, is_fee_payer, maybe_nonce) => {
+                                let is_nonce_account = if is_nonce_account { 1 } else { 0 };
+                                let is_fee_payer = if is_fee_payer { 1 } else { 0 };
+                                let (maybe_nonce, fee_account) =
+                                    if let Some((_, _, fee_account)) = maybe_nonce {
+                                        let fee_account = if fee_account.is_some() { 1 } else { 0 };
+                                        (1, fee_account)
+                                    } else {
+                                        (0, 0)
+                                    };
+
+                                datapoint_info!(
+                                    "store_nonce_ix_error_unexpected",
+                                    ("is_nonce_account", is_nonce_account, i64),
+                                    ("is_fee_payer", is_fee_payer, i64),
+                                    ("maybe_nonce", maybe_nonce, i64),
+                                    ("fee_account", fee_account, i64),
+                                );
+                                continue;
```

Comment:
I really like other `panic!/unreachable!` and similar is removed by this cleaning pr in general.

This might be a personal taste. But retaining `panic!` might be desired for this exceptional situation. I think failing loudly in this case makes more sense. That's because we're pretty sure this wont occur ever in remotely-controllable fashion now. So, if ever this metrics are triggered, this means we started to breaking something in the future modifications. In that case, we'd like to notice this asap. 

Also, I think it's unlikely for lazy engineers to periodically monitor this metrics value... ;)

it's true that this fall-safe/lenient behavior of `continue` improves our cluster's availability but I don't think it adds so much value considering this erroneous condition.

What do you think?

Reply:
I think I agree now.  When I initially wrote this I was pretty tired and second-guessing whether those three cases were in fact the only ones of the dozen or so possibilities that could enter the match.

I think I will replace this commit with a more descriptive `panic!()`